### ISSUE TITLE
ClassCastException during ES operations over HTTPS

### ISSUE THREAD

**Issue by sbouts** [Graylog2/graylog2-server#10045]
## Expected Behavior

The ES operations like indexing, index rollover and other migrations should be performed successfully

## Current Behavior

The ES migration operations (e.g. [this one](https://github.com/Graylog2/graylog2-server/blob/78b817da8001ff8530c4e9e6405a3c20e634b7bd/graylog2-server/src/main/java/org/graylog2/migrations/V20170607164210_MigrateReopenedIndicesToAliases.java#L40)) fail due to a 'java.lang.ClassCastException: [B cannot be cast to java.lang.String' similar to this [graylog issue](https://github.com/Graylog2/graylog2-server/issues/4232)

An example of a full stacktrace is as follows:
```java
2021-02-10T12:22:47.233+01:00 ERROR [ConfigurationManagementPeriodical] Error while running migration <V20170607164210_MigrateReopenedIndicesToAliases{2017-06-07T16:42:10Z}>
java.lang.ClassCastException: [B cannot be cast to java.lang.String
	at org.apache.http.conn.ssl.DefaultHostnameVerifier.getSubjectAltNames(DefaultHostnameVerifier.java:309) ~[graylog.jar:?]
	at org.apache.http.conn.ssl.DefaultHostnameVerifier.verify(DefaultHostnameVerifier.java:112) ~[graylog.jar:?]
	at org.apache.http.conn.ssl.DefaultHostnameVerifier.verify(DefaultHostnameVerifier.java:99) ~[graylog.jar:?]
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.verifyHostname(SSLConnectionSocketFactory.java:463) ~[graylog.jar:?]
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:397) ~[graylog.jar:?]
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:355) ~[graylog.jar:?]
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142) ~[graylog.jar:?]
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:359) ~[graylog.jar:?]
	at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:381) ~[graylog.jar:?]
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:237) ~[graylog.jar:?]
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:185) ~[graylog.jar:?]
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:111) ~[graylog.jar:?]
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185) ~[graylog.jar:?]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83) ~[graylog.jar:?]
	at io.searchbox.client.http.JestHttpClient.executeRequest(JestHttpClient.java:148) ~[?:?]
	at io.searchbox.client.http.JestHttpClient.execute(JestHttpClient.java:77) ~[?:?]
	at org.graylog.storage.elasticsearch6.jest.JestUtils.execute(JestUtils.java:54) ~[?:?]
	at org.graylog.storage.elasticsearch6.jest.JestUtils.execute(JestUtils.java:70) ~[?:?]
	at org.graylog.storage.elasticsearch6.NodeAdapterES6.version(NodeAdapterES6.java:39) ~[?:?]
	at org.graylog2.indexer.cluster.Node.getVersion(Node.java:34) ~[graylog.jar:?]
	at org.graylog2.migrations.V20170607164210_MigrateReopenedIndicesToAliases.getReopenedIndices(V20170607164210_MigrateReopenedIndicesToAliases.java:84) ~[graylog.jar:?]
	at org.graylog2.migrations.V20170607164210_MigrateReopenedIndicesToAliases.getReopenedIndices(V20170607164210_MigrateReopenedIndicesToAliases.java:133) ~[graylog.jar:?]
	at org.graylog2.migrations.V20170607164210_MigrateReopenedIndicesToAliases.lambda$upgrade$0(V20170607164210_MigrateReopenedIndicesToAliases.java:78) ~[graylog.jar:?]
	at java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:269) ~[?:1.8.0_275]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_275]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948) ~[?:1.8.0_275]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_275]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_275]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:1.8.0_275]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:1.8.0_275]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_275]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485) ~[?:1.8.0_275]
	at org.graylog2.migrations.V20170607164210_MigrateReopenedIndicesToAliases.upgrade(V20170607164210_MigrateReopenedIndicesToAliases.java:80) ~[graylog.jar:?]
	at org.graylog2.periodical.ConfigurationManagementPeriodical.doRun(ConfigurationManagementPeriodical.java:42) [graylog.jar:?]
	at org.graylog2.plugin.periodical.Periodical.run(Periodical.java:77) [graylog.jar:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_275]

```

## Possible Solution

Update the graylog [Jest](https://github.com/searchbox-io/Jest) dependency (given that they seem to use the 4.5.3 version [here](https://github.com/searchbox-io/Jest/blob/2d4916e97ca08b36560efef2f784ec130721d654/pom.xml#L71)) and/or update the Apache httpclient dependency

## Steps to Reproduce (for bugs)

1. Setup a ES 6.8.1 HTTPS cluster with a Subject Alternative Name in the certificate
2. Run Graylog 3.3.8 and ingest some logs
3. Migrate to Graylog 4.0.2
3. After starting Graylog 4.0.2 the migrations should fail

## Context

I was trying to accomplish an upgrade from Graylog 3.3.8 to 4.0.2 and this error is keeping Graylog from ingesting data into my ES 6.8.1 cluster

## Your Environment

* Graylog Version: 3.3.8-1 and 4.0.2-1
* Java Version: OpenJDK 1.8.0_275
* Elasticsearch Version: 6.8.1
* MongoDB Version: 4.0.14
* Operating System: Ubuntu 18.04
* Browser version: Firefox 85.0

**Comment by dennisoelkers** [comment]
Hey @sbouts,

thanks for reporting this! Can you poste the special subject alt name of the cert of your ES node(s) that causes http client 4.5.3 to fail?

**Comment by sbouts** [comment]
hey @dennisoelkers, 

First of all, thanks for the quick response!

The server is PEM encoded and the truncated output of `openssl x509 -in <cert> -text` is as follows:
```
X509v3 Subject Alternative Name: 
	DNS:server1.test.local, othername:<unsupported>, othername:<unsupported>
```

**Comment by dennisoelkers** [comment]
Hey @sbouts, that is very useful information. So it is possible that we accidentally reintroduced #4232. I will investigate and possibly fix.

**Comment by sbouts** [comment]
I will try to provide you with a way to create such a certificate. 

The thing I find weird is that the stacktrace I posted shows the error happens on line 309 (which would be [this line in v4.5.3](https://github.com/apache/httpcomponents-client/blob/rel/v4.5.3/httpclient/src/main/java/org/apache/http/conn/ssl/DefaultHostnameVerifier.java#L309)). This error corresponds with the bug report filed for v4.5.3. It still looks to me as if version 4.5.3 is used somehow somewhere given that the [same line in version 4.5.4](https://github.com/apache/httpcomponents-client/blob/rel/v4.5.4/httpclient/src/main/java/org/apache/http/conn/ssl/DefaultHostnameVerifier.java#L309)  is an 'instanceof' check.

Nevertheless I will continue my quest in finding the solution on my server side e.g. by using a different server.

**Comment by sbouts** [comment]
The 'othername unsupported' seems to be due to openssl not supporting othernames as can be seen [here](https://serverfault.com/questions/587662/openssl-always-shows-unsupported-for-all-subjectaltname-othername-utf8-value). I am using version 1.1.1 of OpenSSL

**Comment by sbouts** [comment]
The issue is resolved after upgrading ES to 7.10.x.

I still think the original issue is due to an outdated Jest dependency given that the [Graylog ES 7 implementation](https://github.com/Graylog2/graylog2-server/blob/master/graylog-storage-elasticsearch7/pom.xml) does not make use of said dependency.

**Comment by dennisoelkers** [comment]
Closing it due to being outdated.
